app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,12 @@
+class AdminController < ApplicationController
+  before_action :admin_only
+  before_action :authenticate_user!
```

Comment:
`authenticate_user!` should be called before `admin_only` and rename the `admin_only` method as `verify_admin_user`